### PR TITLE
Update MCVM to typescript

### PR DESCRIPTION
**Pull request by ecuyle**

### CODE REVIEW

File: declarations/MCVersionsManager.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    id: string;
+    type: string;
+    url: string;
+    time: string;
+    releaseTime: string;
+}
+
+export interface VersionsDict {
+    [key: string]: VersionDetails;
+}
```
Comment (JackyLei94) [inline declarations/MCVersionsManager.d.ts]: This works but probably more intuitive as a Record.

```export type VersionsDict = Record<string, VersionDetails>```

See: https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt

Comment (ecuyle) [reply declarations/MCVersionsManager.d.ts]: Oh that's super cool, will update, tyty

File: src/MCServersManager.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ const EULA_TEMPLATE_FILENAME = 'template.eula.txt';
 const EULA_FILENAME = 'eula.txt';
 const SERVER_PROPERTIES_FILENAME = 'server.properties';
 
+console.log(MCVersionsManager);
```
Comment (JackyLei94) [inline src/MCServersManager.js]: delete

Comment (ecuyle) [reply src/MCServersManager.js]: rip

File: src/MCServersManager.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ const EULA_TEMPLATE_FILENAME = 'template.eula.txt';
...
 const mcvm = new MCVersionsManager();
```
Comment (JackyLei94) [inline src/MCServersManager.js]: I think it makes sense here to use your fancy MCVMInterface

**Review by JackyLei94 — APPROVED** [review]
LGTM - nit remove console log before merge

File: src/MCServersManager.js (hunk 3 above)
Comment (ecuyle) [reply src/MCServersManager.js]: Will add when I update MCSM to TS

**Pull request merged**